### ISSUE TITLE
MSVC Version 17.7.3 build issue C1128: number of sections exceeded

### ISSUE THREAD

**Issue by tksuoran** [harfbuzz/harfbuzz#4395]
I am using harfbuzz via cmake fetchcontent:

```cmake
FetchContent_Declare(
    harfbuzz
    GIT_REPOSITORY https://github.com/harfbuzz/harfbuzz.git
    GIT_TAG        main
    GIT_SHALLOW    TRUE
    GIT_PROGRESS   TRUE
)
FetchContent_MakeAvailable(harfbuzz)
```
This causes compilation error:
`src\hb-subset.cc : fatal  error C1128: number of sections exceeded object file format limit: compile with /bigobj`

Replacing `main` with `8.1.1` causes the issue to go away.

The issue _could_ be caused by my project which has some custom cmake flags set.

**Comment by khaledhosny** [comment]
This was fixed for meson in #4632, if you know how to fix it for cmake, please submit a PR.

**Comment by khaledhosny** [comment]
Fixed with #4659
